lib/main.dart

```diff
@@ -16,18 +19,26 @@ class BooksApp extends StatefulWidget {
   State<StatefulWidget> createState() => _BooksAppState();
 }
 
-class _BooksAppState extends State<BooksApp> {
-  BookRouterDelegate _routerDelegate = BookRouterDelegate();
+class _BooksAppState extends State<BooksApp> with RestorationMixin {
+  RestorableBookDelegate _routerDelegate = RestorableBookDelegate();
```

Comment:
The custom RestorableProperty "RestorableBookDelegate" is a wrapper around BookRouterDelegate and responsible for restoring the delegate's state.